src/handlers/proposalsHandlers.ts

```diff
@@ -53,16 +54,22 @@ const getProposals = (
 };
 
 const getProposal = (
-	req: Request<{ id: string }>,
+	req: AuthenticatedRequest,
 	res: Response,
 	next: NextFunction,
 ): void => {
-	const id = Number.parseInt(req.params.id, 10);
-	if (Number.isNaN(id)) {
-		next(new InputValidationError('The entity id must be a number.', []));
+	const { user } = req;
+	const { proposalId } = req.params;
+	if (user === undefined) {
+		next(new FailedMiddlewareError('Unexpected lack of user context.'));
 		return;
 	}
-	loadProposal(id)
+	if (!isId(proposalId)) {
+		next(new InputValidationError('Invalid id parameter.', isId.errors ?? []));
+		return;
+	}
+
+	loadProposal(proposalId, user.id)
```

Comment:
I just want to flag this approach of adding the "createdBy" parameter to the loader itself.  This doesn't deeply couple authorization to the query necessarily, BUT it does risk creating that pattern.

One alternative would be to load the object and THEN check the user access.  That approach might be more prone to timing attacks (e.g. a user could detect if a proposal ID existed vs just not having access based on how long the server took to respond).

Another alternative would be to add something like `checkProposalAuthorization` which is always run *before* loading a proposal regardless of existence (and would return false in either access case).

I opted for this approach for now because it felt more straightforward given the simple nature of authorization that we have today, but honestly it might be better to just start with a `checkProposalAuthorization` pattern since it clearly separates the auth from the object lookup, and is more scalable long term.

Reply:
(I'm probably going to switch to `checkProposalAuthorization` (name TBD)